generate.py

```diff
@@ -543,28 +558,14 @@ def __init__(self, in_directory):
                 # Map issue key to issue data
                 issue_key = issue.issue_key
                 self.issues[issue_key] = issue
-
-                # Sort articles by page number within this issue before assigning indexes
-                sorted_articles = sorted(issue.articles_metadata, key=lambda x: first_page_number(x['pages']))
-
-                for index, article_dict in enumerate(sorted_articles):
-                    # Modify to include issue key directly
-                    article_dict['issue_key'] = issue_key
-                    article_dict['out_filename'] = article_dict['id'] + '.html'
-                    # Assign an index based on sorted order
-                    article_dict['index'] = index
-                    # Assign a pubdate for RSS
-                    article_dict['pubdate'] = article_pubdate(issue, article_dict)
-                    
-                    article = Article(article_dict)
-                    self.articles.append(article)
+                self.articles.extend(issue.articles)
 
     def latest_issue_key(self):
         return max(self.issues.keys(), key=key_to_datetime)
 
     def articles_by_toc_categories(self, toc_categories, issue_key=None):
         filtered_articles = [article for article in self.articles if article.toc_category in toc_categories and (issue_key is None or article.issue_key == issue_key)]
-        return sorted(filtered_articles, key=lambda x: first_page_number(x.pages))
+        return sorted(filtered_articles, key=lambda x: x.first_page_number())
```

Comment:
while you're at it, rename things like `x` into `article`

Reply:
...or follow common Python conventions :D

Reply:
had renamed parts of them to article, but changed back to x - it is shorter and the lambdas are so short that they are understandable enough for now - because all of them are used in sorted() the referencing "what the hell is X?" is explained right there as well

 not sure what the common Python convention is in this case? the sorted() examples in the documentation are mostly descriptive, so should they move into filter methods on the db the names will be more descriptive :D

Reply:
ok